apps/web/src/app/api/webhooks/stripe/route.ts

```diff
@@ -0,0 +1,67 @@
+import { NextRequest, NextResponse } from 'next/server'
+import Stripe from 'stripe'
+import {
+  BadRequestError,
+  UnprocessableEntityError,
+} from '@latitude-data/core/lib/errors'
+import { env } from '@latitude-data/env'
+import { handleSubscriptionUpdate } from '@latitude-data/core/services/billing/handleSubscriptionUpdate'
+import { errorHandler } from '$/middlewares/errorHandler'
+
+const stripeSecretKey = env.STRIPE_SECRET_KEY
+const webhookSecret = env.STRIPE_WEBHOOK_SECRET
+
+if (!stripeSecretKey) {
+  console.error('FATAL: STRIPE_SECRET_KEY environment variable is not set.')
+}
+
+const stripe = stripeSecretKey
+  ? new Stripe(stripeSecretKey, {
+      typescript: true,
+    })
+  : undefined
+
+export const POST = errorHandler(async (req: NextRequest) => {
+  if (!stripe) {
+    throw new UnprocessableEntityError(
+      'Stripe SDK not initialized. Server configuration error.',
+    )
+  }
+
+  if (!webhookSecret) {
+    throw new UnprocessableEntityError(
+      'Stripe webhook secret is not configured. Please set STRIPE_WEBHOOK_SECRET.',
+    )
+  }
+
+  const sig = req.headers.get('stripe-signature')
+  if (!sig) {
+    throw new BadRequestError('Missing Stripe signature header')
+  }
+
+  let event: Stripe.Event
+  try {
+    event = stripe.webhooks.constructEvent(await req.text(), sig, webhookSecret)
+  } catch (error: any) {
+    throw new UnprocessableEntityError(`Webhook Error: ${error.message}`)
+  }
+
+  if (event.type === 'customer.subscription.updated') {
+    const subscription = event.data.object as Stripe.Subscription
```

Comment:
All this signature validation could also be moved to a service and be tested

Reply:
i've added tests to the route